### PR TITLE
Add cucumber and BDD fixtures for Nextest coverage in rust-toy-app

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Restores consistent personalized greeting spelling across BDD scenarios in the rust-toy-app fixture used for Nextest coverage (cucumber-rs).
- Adds rust-toy-app fixture to validate Nextest-based coverage across multi-harness setups (unit, integration, cucumber, BDD rstest, and parametrized rstest scenarios)
- Introduces end-to-end fixture and tests (cucumber-rs 0.21, futures 0.3, rstest) and test harness in cargo toml
- Adds cucumber-based runner rust-toy-app/tests/cucumber.rs and BDD-style tests rust-toy-app/tests/bdd.rs
- Adds feature files and fixtures in rust-toy-app/tests/features (greeting.feature, cli.feature) plus extended rstest fixtures in rust-toy-app/src/lib.rs
- Adds rust-toy-app/README.md documenting usage and how to run tests and generate coverage
- Demonstrates generation of coverage across multiple harnesses merged via generate-coverage action (Nextest mode)
- Guidance for local/testing includes cargo test and cargo llvm-cov nextest commands
- Maintains self-contained fixture; no changes to primary library/action logic

## Changes
- Rust toy fixture and tests
  - Added rust-toy-app as a test fixture to validate coverage collection with multiple harnesses (unit, integration, cucumber, BDDrstest, and parametrized rstest scenarios)
  - New test harnesses and frameworks included: cucumber = 0.21, futures = 0.3
  - [[test]] section added for cucumber-based tests
- Cucumber and BDD tests
  - New cucumber runner: rust-toy-app/tests/cucumber.rs
  - New BDD-style tests: rust-toy-app/tests/bdd.rs
  - Features and scenarios added under rust-toy-app/tests/features/ (greeting.feature, cli.feature)
- Test fixtures and fixtures-driven testing
  - Expanded rust-toy-app/src/lib.rs with extensive rstest fixtures (cli_with_name, cli_default, cli_named) and parametrized scenarios
  - Added scenario-based tests to ensure coverage across edge cases
- Documentation and fixtures
  - rust-toy-app/README.md added with usage, test patterns, and coverage guidance
- Cucumber feature files
  - rust-toy-app/tests/features/greeting.feature and rust-toy-app/tests/features/cli.feature define scenarios for coverage validation
- How this enables Nextest-based coverage
  - Action validates coverage across multiple harnesses using cargo-llvm-cov with Nextest support
  - Fixtures demonstrate how coverage data from different test harnesses can be merged and reported via cobertura
  - Provides ready-to-run fixture for CI to verify generate-coverage action behavior in Nextest mode
- Testing guidance
  - Local verification steps mirror the above; ensure the rust-toy-app fixture builds and its tests generate coverage as described
  - The fixture is self-contained and does not modify primary library/action logic beyond enabling coverage-validation scenarios

## Notes
- This change is self-contained to the test fixture and accompanies the updated generate-coverage action implementation
- The fixture demonstrates generate-coverage action in Nextest mode and illustrates merging coverage across multiple harnesses
- This PR targets the terragon/update-generate-coverage-nextest-6dm7wb branch and aligns with the updated generate-coverage action implementation

📎 **Task**: https://www.terragonlabs.com/task/7e510424-b369-477b-b362-f98f87521d34

📎 **Task**: https://www.terragonlabs.com/task/b39f3f6a-3bce-4b1a-8e7f-7240232fa9e4

### CODE REVIEW

File: rust-toy-app/tests/bdd.rs [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+fn scenario_personalized_greeting_with_name(mut state: GreetingState) {
+    // Given: a name "Alice" is provided
+    state.given_name(Some("Alice".into()));
+
+    // When: the greeting is generated
+    state.when_greeting_generated();
+
+    // Then: the output should be "Hello, Alice!"
+    state.then_greeting_is("Hello, Alice!");
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply rust-toy-app/tests/bdd.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

~~~
Please address the comments from this code review:

## Overall Comments
- In `rust-toy-app/README.md`, the link to the `generate-coverage` action (`[generate-coverage](.github/actions/generate-coverage/)`) is relative to the crate root and likely broken when viewed on GitHub; consider adjusting it to the repository-root-relative path (e.g., `../.github/actions/generate-coverage/` or the full repo URL).
- The `tests/features` directory path is currently hard-coded in multiple places (README examples, `tests/cucumber.rs` main, and the documented action inputs); consider centralizing this path (e.g., via a constant in the test runner and reusing it in docs) to avoid drift if the feature location changes.

## Individual Comments

### Comment 1
<location> `rust-toy-app/tests/cucumber.rs:44-45` </location>
<code_context>
+use rstest::{fixture, rstest};
+use rust_toy_app::cli::Cli;
+
+// =============================================================================
+// Test State (analogous to cucumber's World)
+// =============================================================================
</code_context>

<issue_to_address>
**suggestion (testing):** Add cucumber scenarios that cover the same edge cases as the unit/BDD tests (empty, whitespace, and very long names)

The unit tests in `src/lib.rs` and the BDD rstest suite (`tests/bdd.rs`) already cover empty, whitespace-only, and very long names. Since this fixture is meant to validate cross-harness coverage, please mirror those cases in `tests/features/greeting.feature` so the cucumber harness hits the same paths (e.g., empty name → `"Hello, !"`, whitespace-only name preserved, and a very long name fully included in the greeting). The existing step definitions should suffice; you just need additional scenarios in the feature file.

Suggested implementation:

```rust
// Greeting feature steps (tests/features/greeting.feature)

```

Please update `tests/features/greeting.feature` to add cucumber scenarios that mirror the edge cases already covered in `src/lib.rs` unit tests and `tests/bdd.rs`:

1. **Empty name scenario**

   Add a scenario that uses the existing `Given no name is provided` step and asserts the greeting for an empty name, e.g.:

   ```gherkin
   Scenario: Greeting with an empty name
     Given no name is provided
     When the greeting is generated
     Then the greeting should be "Hello, !"
   ```

   Adjust the `When`/`Then` step texts to match your existing step definitions (for example, `When I ask for a greeting` / `Then I should see "Hello, !"` if that’s how the steps are currently named).

2. **Whitespace-only name scenario**

   Add a scenario that uses `Given a name "<name>" is provided` with a whitespace-only string and asserts that the whitespace is preserved:

   ```gherkin
   Scenario: Greeting with a whitespace-only name
     Given a name "   " is provided
     When the greeting is generated
     Then the greeting should be "Hello,    !"
   ```

   Ensure the number of spaces in the expected greeting matches how your implementation formats the name (e.g., if trimming occurs or not).

3. **Very long name scenario**

   Add a scenario that provides a very long name and verifies it is fully included in the greeting:

   ```gherkin
   Scenario: Greeting with a very long name
     Given a name "This is a very long name that should be fully included in the greeting without truncation" is provided
     When the greeting is generated
     Then the greeting should be "Hello, This is a very long name that should be fully included in the greeting without truncation!"
   ```

4. **Ensure step definitions align**

   The given steps already exist in `tests/cucumber.rs`:

   ```rust
   #[given("no name is provided")]
   fn no_name_provided(world: &mut GreetingWorld) { ... }

   #[given(expr = "a name {string} is provided")]
   fn name_is_provided(world: &mut GreetingWorld, name: String) { ... }
   ```

   Make sure your `When` and `Then` lines in `greeting.feature` use the exact phrases for which you already have `#[when(...)]` and `#[then(...)]` definitions in `tests/cucumber.rs` (for example, `When the greeting is generated` / `Then the greeting should be {string}` or similar). If the phrases differ, either adjust the feature file to match existing steps or add corresponding step definitions in `cucumber.rs`.

With these scenarios in `tests/features/greeting.feature`, the cucumber harness will exercise the same empty, whitespace-only, and very-long-name paths as the existing unit and rstest-based BDD tests.
</issue_to_address>

### Comment 2
<location> `rust-toy-app/README.md:9` </location>
<code_context>
+
+## Purpose
+
+This simple CLI application demonstrates various Rust testing patterns for
+coverage collection validation:
+
</code_context>

<issue_to_address>
**issue (review_instructions):** "CLI" is used here without being expanded on first use; expand it (for example, "Command Line Interface (CLI)").

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.md`

**Instructions:**
Define uncommon acronyms on first use.

</details>
</issue_to_address>

### Comment 3
<location> `rust-toy-app/README.md:16` </location>
<code_context>
+|-----------|-----------|---------|---------|
+| Unit tests | [rstest](https://crates.io/crates/rstest) | `src/lib.rs` | Standard |
+| Integration tests | rstest + assert_cmd | `tests/cli.rs` | Standard |
+| BDD (Gherkin) | [cucumber-rs](https://cucumber-rs.github.io/cucumber/current/) | `tests/cucumber.rs`, `tests/features/*.feature` | Custom (`harness = false`) |
+| BDD (rstest-style) | rstest | `tests/bdd.rs` | Standard |
+
</code_context>

<issue_to_address>
**issue (review_instructions):** "BDD" is introduced without definition; expand it on first use (for example, "Behaviour-Driven Development (BDD)").

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.md`

**Instructions:**
Define uncommon acronyms on first use.

</details>
</issue_to_address>

### Comment 4
<location> `rust-toy-app/README.md:118` </location>
<code_context>
+    bdd.rs          # BDD-style rstest tests (standard harness)
+    features/
+      greeting.feature  # Gherkin scenarios for greeting
+      cli.feature       # Gherkin scenarios for CLI behavior
+```
+
</code_context>

<issue_to_address>
**issue (review_instructions):** "behavior" uses US spelling; switch to en-GB spelling "behaviour".

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.md`

**Instructions:**
Use en-GB-oxendic (-ize / -yse / -our) spelling and grammar.

</details>
</issue_to_address>
~~~

File: rust-toy-app/tests/bdd.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (leynos) [reply rust-toy-app/tests/bdd.rs]: @coderabbitai "Personalized" is correct for en-GB-oxendict (-ize / -yse / -our)
